haystack/nodes/llm/prompt_node.py

```diff
@@ -0,0 +1,702 @@
+        default_prompt_template: Optional[Union[str, PromptTemplate]] = None,
+        output_variable: Optional[str] = None,
+        max_length: Optional[int] = 100,
+        api_key: Optional[str] = None,
+        use_auth_token: Optional[Union[str, bool]] = None,
+        use_gpu: Optional[bool] = None,
+        devices: Optional[List[Union[str, torch.device]]] = None,
+    ):
+        super().__init__()
+        self.default_prompt_template: Union[str, PromptTemplate, None] = default_prompt_template
+        self.output_variable: Optional[str] = output_variable
+        self.model_name_or_path: Union[str, PromptModel] = model_name_or_path
+        self.prompt_model: PromptModel
+        if self.default_prompt_template is not None and not self.is_supported_template(self.default_prompt_template):
+            raise ValueError(
+                f"Prompt template {default_prompt_template} is not supported. "
+                f"Please select one of: {self.get_prompt_template_names()}"
+            )
+
+        if isinstance(model_name_or_path, str):
+            self.prompt_model = PromptModel(
+                model_name_or_path=model_name_or_path,
+                max_length=max_length,
+                api_key=api_key,
+                use_auth_token=use_auth_token,
+                use_gpu=use_gpu,
+                devices=devices,
+            )
+        elif isinstance(model_name_or_path, PromptModel):
+            self.prompt_model = model_name_or_path
+        else:
+            raise ValueError(f"model_name_or_path must be either a string or a PromptModel object")
+
+    def __call__(self, *args, **kwargs) -> List[str]:
+        """
+        This method is invoked when the component is called directly, for example:
+        ```python
+            PromptNode pn = ...
+            sa = pn.set_default_prompt_template("sentiment-analysis")
+            sa(documents=[Document("I am in love and I feel great!")])
+        ```
+        """
+        if "prompt_template_name" in kwargs:
+            prompt_template_name = kwargs["prompt_template_name"]
+            kwargs.pop("prompt_template_name")
+            return self.prompt(prompt_template_name, *args, **kwargs)
+        else:
+            return self.prompt(self.default_prompt_template, *args, **kwargs)
+
+    def prompt(self, prompt_template: Optional[Union[str, PromptTemplate]], *args, **kwargs) -> List[str]:
+        """
+        Prompts the model and represents the central API for the PromptNode. It takes a prompt template,
+        a list of non-keyword and keyword arguments, and returns a list of strings - the responses from
+        the underlying model.
+
+        The optional prompt_template parameter, if specified, takes precedence over the default prompt
+        template for this PromptNode.
+
+        :param prompt_template: The name of the optional prompt template to use
+        :type prompt_template: Optional[Union[str, PromptTemplate]]
```

Comment:
:type description is not needed here because we already have a type annotation in the method's signature.